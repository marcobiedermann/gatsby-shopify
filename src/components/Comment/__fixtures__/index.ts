import { CommentProps } from '..';
import author from '../../Author/__fixtures__';

const comment: CommentProps = {
  author,
  contentHtml: '',
  shopifyId: '12345678-9012-3456-7890-123456789012',
};

export default comment;
