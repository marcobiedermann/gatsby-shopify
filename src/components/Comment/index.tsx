import React, { FC } from 'react';
import Author, { AuthorProps } from '../Author';

export interface CommentProps {
  author: AuthorProps;
  contentHtml: string;
  shopifyId: string;
}

const Comment: FC<CommentProps> = (props) => {
  const { contentHtml, author } = props;

  return (
    <div>
      <div dangerouslySetInnerHTML={{ __html: contentHtml }} />
      <Author {...author} />
    </div>
  );
};

export default Comment;
