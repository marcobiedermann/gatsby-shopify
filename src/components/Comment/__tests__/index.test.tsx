import { render } from '@testing-library/react';
import React from 'react';
import Comment from '..';
import comment from '../__fixtures__';

describe('comment component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Comment {...comment} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
