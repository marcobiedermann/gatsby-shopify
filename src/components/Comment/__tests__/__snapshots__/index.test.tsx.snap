// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div />
    <div>
      <span>
        John Doe
      </span>
    </div>
  </div>
</DocumentFragment>
`;
